python/private/repo_utils.bzl

```diff
@@ -341,25 +341,60 @@ def _outputs_to_str(result):
             lines.append("<{} empty>".format(name))
     return "\n".join(lines)
 
+# This includes the vendored _translate_cpu and _translate_os from
+# @platforms//host:extension.bzl at version 0.0.9 so that we don't
+# force the users to depend on it.
+
 def _get_platforms_os_name(rctx):
     """Return the name in @platforms//os for the host os.
 
     Args:
         rctx: repository_ctx
 
     Returns:
-        `str | None`. The target name if it maps to known platforms
-        value, otherwise None.
+        `str`. The target name.
     """
     os = rctx.os.name.lower()
-    if "linux" in os:
-        return os
-    if "windows" in os:
-        return "windows"
-    if "mac" in os:
+
+    if os.startswith("mac os"):
         return "osx"
+    if os.startswith("freebsd"):
+        return "freebsd"
+    if os.startswith("openbsd"):
+        return "openbsd"
+    if os.startswith("linux"):
+        return "linux"
+    if os.startswith("windows"):
+        return "windows"
+    return os
+
+def _get_platforms_arch_name(rctx):
```

Comment:
nit: maybe `get_platforms_cpu_name`, instead? The result is intended to be used with the `@platforms//cpu:<name>` string